### PR TITLE
Add support for Philips Hue Smart Button

### PR DESCRIPTION
**Pull request by dljsjr**
Generalize the work done to support the multi-button remotes to the single button smart button.

### CODE REVIEW

File: drivers/SmartThings/philips-hue/profiles/hue-smart-button.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  - id: main
+    capabilities:
+      - id: button
+        version: 1
+      - id: battery
+        version: 1
+      - id: refresh
+        version: 1
+    categories:
+    - name: RemoteController
```
Comment (tpmanley) [inline drivers/SmartThings/philips-hue/profiles/hue-smart-button.yml:12]: Let's use `Button`

File: drivers/SmartThings/philips-hue/src/disco/button.lua [hunk 2]
```diff
@@ -121,10 +121,16 @@ function M.handle_discovered_device(
       )
       return
     end
+
     local button_profile_ref = ""
-    if button_description.num_buttons == 4 then
+    -- For Philips Hue Smart Button device which contains only 1 button
+    if button_description.num_buttons == 1 then
+      button_profile_ref = "HueSmartButton"
+      -- For Philips Hue Dimmer Remote which contains 4 buttons
```
Comment (tpmanley) [inline drivers/SmartThings/philips-hue/src/disco/button.lua:129]: nit: Also the Tap Dial Switch has 4 buttons

File: drivers/SmartThings/philips-hue/src/hue_device_types.lua [hunk 3]
```diff
@@ -13,7 +13,8 @@ local HueDeviceTypes = {
 }
 
 local SupportedNumberOfButtons = {
-  [4] = true
+  [1] = true, -- For Philips Hue Smart Button device which contains only 1 button
+  [4] = true, -- For Philips Hue Dimmer Remote which contains 4 buttons
```
Comment (tpmanley) [inline drivers/SmartThings/philips-hue/src/hue_device_types.lua:17]: nit: and Tap Dial Switch

File: drivers/SmartThings/philips-hue/src/disco/button.lua (hunk 2 above)
Comment (tpmanley) [reply drivers/SmartThings/philips-hue/src/disco/button.lua:129]: This is already fixed in the next PR

File: drivers/SmartThings/philips-hue/profiles/hue-smart-button.yml (hunk 1 above)
Comment (dljsjr) [reply drivers/SmartThings/philips-hue/profiles/hue-smart-button.yml:12]: For all buttons or just the single button/smart button?

Comment (dljsjr) [reply drivers/SmartThings/philips-hue/profiles/hue-smart-button.yml:12]: Adjusted for single button on https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1359, lmk if it should be done for the multi-buttons as well.

**Review by NoahCornell — APPROVED** [review]

File: drivers/SmartThings/philips-hue/profiles/hue-smart-button.yml (hunk 1 above)
Comment (tpmanley) [reply drivers/SmartThings/philips-hue/profiles/hue-smart-button.yml:12]: Looks good. We've been sticking with RemoteControl for more than one button

**Review by cjswedes — APPROVED** [review]

**Pull request merged**